app/src/main/java/com/google/samples/apps/sunflower/viewmodels/GalleryViewModel.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.sunflower.viewmodels
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.asLiveData
+import androidx.lifecycle.liveData
+import androidx.lifecycle.switchMap
+import com.google.samples.apps.sunflower.api.UnsplashService
+import com.google.samples.apps.sunflower.data.UnsplashRepository
+import com.google.samples.apps.sunflower.data.UnsplashSearchResult
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+
+@ExperimentalCoroutinesApi
+@FlowPreview
+class GalleryViewModel internal constructor(
+    repository: UnsplashRepository
+) : ViewModel() {
+
+    private val queryLiveData = MutableLiveData<String>()
+
+    val repoResult: LiveData<UnsplashSearchResult> = queryLiveData.switchMap { queryString ->
+        liveData {
+            val repos = repository.getSearchResultStream(queryString).asLiveData(Dispatchers.Main)
+            emitSource(repos)
```

Comment:
Do you need `Dispatchers.Main` here? I think `asLiveData()` should be enough

Reply:
Done